mirrord/operator/src/client.rs

```diff
@@ -187,6 +218,26 @@ impl OperatorApi {
             return Ok(None);
         };
 
+        // Warns the user if their license is close to expiring.
+        //
+        // I(alex) considered doing a check for validity also here for expired licenses,
+        // but maybe the time of the local user and of the operator are out of sync, so we
+        // could end up blocking a valid license (or even just warning on it could be
+        // confusing).
+        if let Some(expiring_soon) =
+            LicenseValidity::from(operator.spec.license.expire_at).close_to_expiring()
+        {
+            if expiring_soon >= 0 {
+                let expiring_message = format!(
+                    "Operator license will expire soon, in {} days!",
+                    expiring_soon,
+                );
+
+                progress.warning(&expiring_message);
+                warn!(expiring_message);
+            }
+        }
```

Comment:
Do you know if exisiting operator installations check expired licenses? If not, maybe we should add a check for this somewhere here and return an error (while making sure that endpoints like `status` are still accessible)? Ofc this check could be bypassed, but we can't do anything about it

Reply:
As far as I remember, there are no checks for license expiration.

We could check here, but wouldn't this break backwards compatibility? Like, if you update mirrord, but do not update the operator, now mirrord will perform this check and say that your license is invalid, even though the operator would allow you to continue.

Also, I think this would be kinda redundant, isn't it better to just have 1 place of authority (the operator itself in the case of these PRs)? We can talk more about how tight up we want this to be on discord.